src/System.Runtime.Extensions/tests/System/Net/WebUtility.cs

```diff
@@ -46,11 +50,80 @@ public static void HtmlEncode(string value, string expected)
             Assert.Equal(expected, WebUtility.HtmlEncode(value));
         }
 
-        public static IEnumerable<object[]> UrlDecode_TestData()
+        // Shared test data for UrlEncode + Decode and their ToBytes counterparts
+
+        public static IEnumerable<Tuple<string, string>> UrlDecode_SharedTestData()
+        {
+            // Recent change brings function inline with RFC 3986 to return hex-encoded chars in uppercase
+            yield return Tuple.Create("%2F%5C%22%09Hello!+%E2%99%A5%3F%2F%5C%22%09World!+%E2%99%A5%3F%E2%99%A5", "/\\\"\tHello! \u2665?/\\\"\tWorld! \u2665?\u2665");
+            yield return Tuple.Create("Hello, world", "Hello, world"); // No special chars
+            yield return Tuple.Create("%F0%90%8F%BF", "\uD800\uDFFF"); // Surrogate pair
+        }
+
+        [ActiveIssue(7166)]
```

Comment:
@davidsh Sorry for my earlier comment, I was mistaken. This is actually test data (it's being used as `MemberData`), not a test itself, so I don't think `Fact`/`Theory` applies here. I'll remove the `ActiveIssue` though, to prevent more confusion.